operator crashes periodically with "panic: close of closed channel"

I noticed the rook-operator running master-latest (v0.3.1-99-gea2b435) had crashed a few times in the past couple hours:
```
> kubectl get pod
NAME                             READY     STATUS    RESTARTS   AGE
rook-operator-1641293690-h77mx   1/1       Running   3          2h
```

The logs show the reason for the crash:
```
> kubectl logs --previous=true rook-operator-1641293690-h77mx
<ommitted>
2017-04-28 02:17:56.987758 W | operator: cancelling cluster tpr watch. failed to poll cluster event. EOF
2017-04-28 02:17:56.988055 E | operator: failed to watch clusters. received invalid event from API server: failed to poll cluster event. EOF
2017-04-28 02:17:56.988136 I | op-cluster: Stopping monitoring of cluster rook in namespace rook
panic: close of closed channel

goroutine 86 [running]:
github.com/rook/rook/pkg/operator.(*clusterManager).stopTrack(0xc4200cfe60, 0xc42040c3c0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/cluster.go:128 +0x1a1
github.com/rook/rook/pkg/operator.(*clusterManager).startCluster.func1(0xc4200cfe60, 0xc42040c3c0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/cluster.go:186 +0x2e7
created by github.com/rook/rook/pkg/operator.(*clusterManager).startCluster
	/home/rook/go/src/github.com/rook/rook/pkg/operator/cluster.go:186 +0x20c
```

Note that a very similar event happens for the pool TPR (EOF), but we recover from that one OK:
```
2017-04-28 02:01:12.361612 E | operator: failed to watch rook pool tpr. received invalid event from API server: failed to poll pool event. EOF. retrying...
2017-04-28 02:01:18.362424 I | operator: finding existing pools...
2017-04-28 02:01:18.366234 I | operator: found 1 pools. ensuring they exist.
2017-04-28 02:01:18.366317 I | operator: checking pool replicapool in namespace rook
2017-04-28 02:01:18.376719 I | op-cluster: pool replicapool already exists in namespace rook 
2017-04-28 02:01:18.376790 I | operator: start watching pool tpr in namespace rook: 11451
```

--- a/pkg/operator/tracker.go
+++ b/pkg/operator/tracker.go
@@ -21,9 +21,9 @@ package operator
 import "sync"
 
 type tprTracker struct {
-	stopChMap   map[string]chan struct{}
-	clusterRVs  map[string]string
-	waitCluster sync.WaitGroup
+	stopChMap  map[string]chan struct{}
+	clusterRVs map[string]string
+	sync.RWMutex
 }
 
 func newTPRTracker() *tprTracker {
@@ -43,6 +43,8 @@ func (t *tprTracker) add(name, version string) {
 func (t *tprTracker) remove(name string) {
 	delete(t.clusterRVs, name)
 
+	t.Lock()
+	defer t.Unlock()
 	if stopCh, ok := t.stopChMap[name]; ok {
 		close(stopCh)
 	}
@@ -50,8 +52,10 @@ func (t *tprTracker) remove(name string) {
 }
 
 func (t *tprTracker) stop() {
+	t.Lock()
+	defer t.Unlock()
 	for _, stop := range t.stopChMap {
 		close(stop)
 	}
-	t.waitCluster.Wait()
+	t.stopChMap = map[string]chan struct{}{}
 }

